Userland/Applications/ImageViewer/ViewWidget.h

```diff
@@ -34,8 +35,16 @@ class ViewWidget final : public GUI::Frame {
     int toolbar_height() { return m_toolbar_height; }
     bool scaled_for_first_image() { return m_scaled_for_first_image; }
     void set_scaled_for_first_image(bool val) { m_scaled_for_first_image = val; }
+    void set_path_for_first_image(const String& path)
+    {
+        m_path = path;
+        m_current_index = current_index();
```

Comment:
This seems a little silly, `current_index()` already assigns to `m_current_index`.
Why does this need to exist anyway? the next user will call `current_index` and get this behaviour, no?

Reply:
Yes, it does seem like that, but the file being opened, should really update the `m_current_index`. otherwise, the value for the first file opened never get set.

Reply:
as we are getting index from the `m_path` so, first we set `m_path` and the we explicitly update `m_current_path` to the index of `m_path`. Make sense?

Reply:
as `current_index` is already assigning value to `current_index`, so instead of assigning here, we can just call `current_index` here, to set the initial value for the index of this file.

Reply:
Then perhaps this function should set that first file value, and that getter shouldn't touch it at all.

Reply:
yes, that's exactly what is happening. this function set it. and the navigation actions update them.

Reply:
it can be written as this:

   ``` void set_path_for_first_image(const String& path)
    {
        m_path = path;
        current_index();
```